### PR TITLE
Inverse of Cholesky decomposition

### LINKED ISSUE #847: Inverse of static matrix from cholesky decomposition fails
I get the following output (using 970ce8b9cba74f6a8a35cd5f414d143462b7a5ba):
```
julia> A = 2 * MMatrix{3,3, Float64}(I)
3×3 MArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
 2.0  0.0  0.0
 0.0  2.0  0.0
 0.0  0.0  2.0

julia> cholesky(A)
Cholesky{Float64,MArray{Tuple{3,3},Float64,2,9}}
U factor:
3×3 UpperTriangular{Float64,MArray{Tuple{3,3},Float64,2,9}}:
 1.41421  0.0      0.0
  ⋅       1.41421  0.0
  ⋅        ⋅       1.41421

julia> inv(cholesky(A))
3×3 Array{Float64,2}:
 0.5  0.0  0.0
 0.0  0.5  0.0
 0.0  0.0  0.5

julia> B = 2 * SMatrix{3,3, Float64}(I)
3×3 SArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
 2.0  0.0  0.0
 0.0  2.0  0.0
 0.0  0.0  2.0

julia> inv(cholesky(B))
ERROR: MethodError: no method matching strides(::SArray{Tuple{3,3},Float64,2,9})
Closest candidates are:
  strides(::SubArray) at subarray.jl:331
  strides(::Union{Base.ReinterpretArray{T,N,S,A} where S where A<:Union{SubArray{T,N,A,I,true} where I<:Union{Tuple{Vararg{Real,N} where N}, Tuple{AbstractUnitRange,Vararg{Any,N} where N}} where A<:DenseArray where N where T, DenseArray} where N where T, Base.ReshapedArray{T,N,A,MI} where MI<:Tuple{Vararg{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64},N} where N} where A<:Union{Base.ReinterpretArray{T,N,S,A} where S where A<:Union{SubArray{T,N,A,I,true} where I<:Union{Tuple{Vararg{Real,N} where N}, Tuple{AbstractUnitRange,Vararg{Any,N} where N}} where A<:DenseArray where N where T, DenseArray} where N where T, SubArray{T,N,A,I,true} where I<:Union{Tuple{Vararg{Real,N} where N}, Tuple{AbstractUnitRange,Vararg{Any,N} where N}} where A<:DenseArray where N where T, DenseArray} where N where T, DenseArray}) at reinterpretarray.jl:76
  strides(::PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm) where {T, N, perm} at permuteddimsarray.jl:63
  ...
Stacktrace:
 [1] stride(::SArray{Tuple{3,3},Float64,2,9}, ::Int64) at ./abstractarray.jl:396
 [2] stride1(::SArray{Tuple{3,3},Float64,2,9}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/LinearAlgebra.jl:197
 [3] _chkstride1 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/LinearAlgebra.jl:203 [inlined]
 [4] chkstride1 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/LinearAlgebra.jl:201 [inlined]
 [5] potrs!(::Char, ::SArray{Tuple{3,3},Float64,2,9}, ::Array{Float64,2}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/lapack.jl:3097
 [6] ldiv!(::Cholesky{Float64,SArray{Tuple{3,3},Float64,2,9}}, ::Array{Float64,2}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/cholesky.jl:470
 [7] inv(::Cholesky{Float64,SArray{Tuple{3,3},Float64,2,9}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/factorization.jl:59
 [8] top-level scope at none:1
```

Getting the inverse using the cholesky is useful for covariance manipulations.

### PR DESCRIPTION
**Pull request by mateuszbaran**
This should fix #847.

### CODE REVIEW

**Comment by mateuszbaran** [comment]
I guess I could also add division by Cholesky decomposition.

File: test/chol.jl [hunk 1]
```diff
@@ -48,6 +48,43 @@ using LinearAlgebra: PosDefException
...
+            d = SMatrix{3,3}(d_a)
+
+            @test (@inferred c \ d) isa SMatrix{3,3,elty}
+            @test c \ d ≈ c_a \ d_a
+            @test (@inferred c \ Symmetric(d)) isa SMatrix{3,3,elty}
+            @test c \ Symmetric(d) ≈ c_a \ Symmetric(d_a)
+
+            if VERSION >= v"1.3"
+                @test (@inferred d / c) isa SMatrix{3,3,elty}
+                @test d / c ≈ d_a / c_a
```
Comment (c42f) [inline test/chol.jl:78]: I assume this still works for < v"1.3" and it's the `@inferred` which fail?  In which case the value tests could be moved outside this block.

**Review by c42f — APPROVED** [review]
Looks good to me.

File: test/chol.jl (hunk 1 above)
Comment (mateuszbaran) [reply test/chol.jl:78]: No, it's actually the stdlib `Array` code that fails here.

**Review by mschauer — APPROVED** [review]

File: test/chol.jl (hunk 1 above)
Comment (c42f) [reply test/chol.jl:78]: Oh right, worth a short comment?

Comment (mateuszbaran) [reply test/chol.jl:78]: Sure, I'll write it 🙂

**Pull request merged**